### PR TITLE
tap_esc: changes needed to run on qurt

### PR DESCRIPTION
**Pull request by ChristophTobler**

### CODE REVIEW

File: src/drivers/tap_esc/tap_esc.cpp [hunk 1]
```diff
@@ -852,7 +852,7 @@ int TAP_ESC::control_callback(uint8_t control_group, uint8_t control_index, floa
 }
 
 int
-TAP_ESC::ioctl(file *filp, int cmd, unsigned long arg)
+TAP_ESC::ioctl(FILE *filp, int cmd, unsigned long arg)
```
Comment (bkueng) [inline src/drivers/tap_esc/tap_esc.cpp]: From what I see in CDev, this should be `file_t`

File: src/drivers/tap_esc/tap_esc.cpp [hunk 2]
```diff
@@ -869,7 +869,7 @@ TAP_ESC::ioctl(file *filp, int cmd, unsigned long arg)
 
 	case MIXERIOCLOADBUF: {
 			const char *buf = (const char *)arg;
-			unsigned buflen = strnlen(buf, 1024);
+			unsigned buflen = strlen(buf);
```
Comment (Lampoo) [inline src/drivers/tap_esc/tap_esc.cpp]: I do see there is a strnlen version in qurt, any reason why it is needed for qurt?

Comment (ChristophTobler) [reply src/drivers/tap_esc/tap_esc.cpp]: @Lampoo Can you point me to that source?

File: src/drivers/tap_esc/tap_esc.cpp (hunk 1 above)
Comment (ChristophTobler) [reply src/drivers/tap_esc/tap_esc.cpp]: You are right

**Comment by ChristophTobler** [comment]
@bkueng Bench tested on an Aero

File: src/drivers/tap_esc/tap_esc.cpp (hunk 2 above)
Comment (Lampoo) [reply src/drivers/tap_esc/tap_esc.cpp]: @ChristophTobler I can see it is at :src/platforms/qurt/px4_layer/px4_qurt_impl.cpp:148:size_t strnlen(const char *s, size_t maxlen)

**Pull request merged**